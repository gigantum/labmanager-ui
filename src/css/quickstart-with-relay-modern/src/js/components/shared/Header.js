import React, { Component } from 'react'
import GigantumLogo from '../../../images/logos/gigantum.png';

export default class Header extends Component {
  render() {
    return (
      <div className={'header__container flex flex--row justify--space-between'}>
        <div className={'flex-1-0-auto'}>
          <h1 className='header__title text-center'>Gigantum</h1>
        </div>
        <div>
          <img alt='gigantum logo' src={GigantumLogo} height='60'/>
        </div>
      </div>
    )
  }
}
