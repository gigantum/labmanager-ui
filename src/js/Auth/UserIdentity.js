//vendor
import {
  graphql,
} from 'react-relay'
//mutations
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//environment
import {fetchQuery} from 'JS/createRelayEnvironment';

const userIdentityQuery = graphql`
  query UserIdentityQuery{
    userIdentity{
      id
      username
      email
      givenName
      familyName
      isSessionValid
    }
  }
`;

const UserIdentity = {
  getUserIdentity: () => {

    return new Promise((resolve, reject) =>{

      let fetchData = function(){

        fetchQuery(userIdentityQuery(), {}).then((response, error) => {

          if(response){
            resolve(response)
          }else{

            reject(response)
          }
        }).catch((error) =>{

          reject(error)
        })
      }

      fetchData()
    })
  },
  removeUserIdentity: () => {
    RemoveUserIdentityMutation(()=>{
      console.log('removed user')
    })
  }

}

export default UserIdentity
