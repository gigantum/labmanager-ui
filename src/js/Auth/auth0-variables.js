export const AUTH_CONFIG = {
  domain: 'calum.auth0.com',
  clientId: '9wFN5CZA4Z9xa1r8KZIZbevpjHG7bAab',
  callbackUrl: window.location.origin + '/callback'
}
