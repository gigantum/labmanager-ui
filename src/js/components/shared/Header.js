import React, { Component } from 'react';
import { Link } from 'react-router-dom';
// components
import User from './User';


export default class Header extends Component {
  /**
    @param {}
    logout through Auth0
  */
  logout() {
    this.props.auth.logout();
  }
  render() {
    const { isAuthenticated } = this.props.auth;
    return (
      <div className="Header flex flex--row justify--space-between">
        <div className="flex justify--space-around flex-1-0-auto">

          <ul className="Header__nav flex flex--row justify--space-between">
            <li>
              <Link
                className="Header__nav-item Header__nav-item--datasets flex flex--row justify--space-between"
                to={{ pathname: '/datasets' }}
              >
                <div className="Header__datasets-icon" />
                Datasets
              </Link>
            </li>
            <li>
              <Link
                className="Header__nav-item Header__nav-item--labbooks flex flex--row justify--space-between"
                to={{ pathname: '/projects' }}
              >
                <div className="Header__labbook-icon" />
                Projects
              </Link>
            </li>

          </ul>

          {
            isAuthenticated() && (
            <User {...this.props} />
              )
          }
        </div>
      </div>
    );
  }
}
