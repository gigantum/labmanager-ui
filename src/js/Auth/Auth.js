import history from 'JS/history';
import auth0 from 'auth0-js';
import { AUTH_CONFIG } from './auth0-variables';
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//store
import store from 'JS/redux/store'


export default class Auth {
  auth0 = new auth0.WebAuth({
    domain: AUTH_CONFIG.domain,
    clientID: AUTH_CONFIG.clientId,
    redirectUri: AUTH_CONFIG.callbackUrl,
    audience:  AUTH_CONFIG.audience,
    responseType: 'token id_token',
    scope: 'openid profile email user_metadata'
  });

  constructor() {
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.handleAuthentication = this.handleAuthentication.bind(this);
    this.isAuthenticated = this.isAuthenticated.bind(this);

  }

  login() {
    this.auth0.authorize();
  }

  handleAuthentication() {

    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {

        this.setSession(authResult);

      } else if (err) {
        console.error(err);
      //  alert(`Error: ${err.error}. Check the console for further details.`);
      }
    });
  }

  setSession(authResult) {

    // Set the time that the access token will expire at
    let expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
    localStorage.setItem('access_token', authResult.accessToken);
    localStorage.setItem('id_token', authResult.idToken);
    localStorage.setItem('expires_at', expiresAt);
    localStorage.setItem('family_name', authResult.idTokenPayload.family_name);
    localStorage.setItem('given_name', authResult.idTokenPayload.given_name);
    localStorage.setItem('email', authResult.idTokenPayload.email);
    localStorage.setItem('username', authResult.idTokenPayload.nickname);
    //redirect to labbooks when user logs in

    let storrageRoute = sessionStorage.getItem('CALLBACK_ROUTE')
    let route = storrageRoute !== '' ? storrageRoute : `/labbooks`

    history.replace(route)
  }

  logout() {

    localStorage.removeItem('access_token');
    localStorage.removeItem('id_token');
    localStorage.removeItem('expires_at');
    localStorage.removeItem('family_name')
    localStorage.removeItem('given_name');
    localStorage.removeItem('email');
    localStorage.removeItem('username');
    RemoveUserIdentityMutation(()=>{
      //redirect to root when user logs out
      history.replace('/');
    })
  }

  isAuthenticated() {
    // Check whether the current time is past the
    // access token's expiry time
    let expiresAt = JSON.parse(localStorage.getItem('expires_at'));

    return new Date().getTime() < expiresAt;
  }
}
