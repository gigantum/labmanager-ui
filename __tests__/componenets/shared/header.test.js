import React, { Component } from 'react'
import {MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import Header from 'Components/shared/Header';
import Auth from 'JS/Auth/Auth';
import history from 'JS/history';

const auth = new Auth();
auth.isAuthenticated = function(){return false};

test('Test Header rendering', () => {
  const component = renderer.create(
    <MemoryRouter history={history}>
      <Header auth={auth} history={history}/>
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
