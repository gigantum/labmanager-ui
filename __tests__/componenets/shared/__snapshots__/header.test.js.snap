// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header rendering 1`] = `
<div
  className="Header flex flex--row justify--space-between"
>
  <div
    className="flex justify--space-around flex-1-0-auto"
  >
    <ul
      className="Header__nav flex flex--row justify--space-between"
    >
      <li>
        <a
          className="Header__nav-item Header__nav-item--datasets flex flex--row justify--space-between"
          href="/datasets"
          onClick={[Function]}
        >
          <div
            className="Header__datasets-icon"
          />
          Datasets
        </a>
      </li>
      <li>
        <a
          className="Header__nav-item Header__nav-item--labbooks flex flex--row justify--space-between"
          href="/labbooks"
          onClick={[Function]}
        >
          <div
            className="Header__labbook-icon"
          />
          Lab Books
        </a>
      </li>
    </ul>
  </div>
</div>
`;
